index.js

```diff
@@ -4109,14 +4109,19 @@ export const trimComponents = (components, format) => {
   const keyCache = {};
   const filteredComponents = [];
   for (const comp of components) {
+    if (!comp.version) continue;
```

Comment:
Can we do ```comp.version = comp.version || ""```

Reply:
Well I think needed to add this because the tool i was using to read the SBoM file did complain that some purl did not have a version. But this seem to be an issue with the tool I was using. I'm reverting the last change.